### ISSUE TITLE
Make it easy to compare DataFrames that are the same, but with different column ordering

### ISSUE THREAD

**Issue by MrPowers** [mrpowers-io/spark-fast-tests#79]
See this question: https://github.com/MrPowers/spark-fast-tests/issues/78

There's already an option to compare DataFrame that are equal, but have different row ordering.  This feature is to compare DataFrames that have different column orderings.

**Comment by akuzni2** [comment]
I'd actually like to help with this feature if possible as it would come in handy for my team.

**Comment by MrPowers** [comment]
@akuzni2 - cool, feel free to send a PR ;)

Here's the current syntax for unordered row comparisons: `assertSmallDataFrameEquality(sourceDF, expectedDF, orderedComparison = false)`.

What do you think about this syntax for unordered Column comparisons? `assertSmallDataFrameEquality(sourceDF, expectedDF, orderedColumnComparison = false)`

It should be relatively straightforward to order the columns alphabetically and then perform the comparison.

Let me know if you feel up for creating this PR.  If not, I'm happy to add this feature.  Just let me know either way, thanks!

**Comment by akuzni2** [comment]
@MrPowers - I'd like to take a look at it if you don't mind :) that might get me on a roll to help contribute to other things here

**Comment by MrPowers** [comment]
@akuzni2 - sounds great, feel free to open up a pull request and let me know if you have any questions / need any help!

**Comment by akuzni2** [comment]
Hey @MrPowers  I'm actually struggling to built the project. I've noticed in the github CI pipeline you're using JRE 8u275 and SBT 1.3.13. I've tried to build a docker image to install those specific things (so I don't mess up my local setup) however getting issues with the build.sbt file. I keep getting an error 
```
/build.sbt:1: error: not found: value GitVersioning
enablePlugins(GitVersioning)
```

Is there any special needs for building this project?

Heres my Dockerfile which creates a build environment, installs sbt and try's to run `sbt test` on the project
```
ARG OPENJDK_TAG=8u275-jre-slim
FROM openjdk:${OPENJDK_TAG}

ARG SBT_VERSION=1.3.13

RUN apt-get update && apt-get -qq -y install curl

# Install sbt
RUN \
  mkdir /working/ && \
  cd /working/ && \
  curl -L -o sbt-$SBT_VERSION.deb https://dl.bintray.com/sbt/debian/sbt-$SBT_VERSION.deb && \
  dpkg -i sbt-$SBT_VERSION.deb && \
  rm sbt-$SBT_VERSION.deb && \
  apt-get update && \
  apt-get install sbt && \
  cd && \
  rm -r /working/ && \
  sbt sbtVersion

COPY * /

RUN sbt -Dspark.testVersion=2.4.0 ++2.12.12 test
```

**Comment by MrPowers** [comment]
@akuzni2 - You can manage the Java versions on your machine with [jenv](https://mungingdata.com/java/jenv-multiple-versions-java/) or [sdkman](https://mungingdata.com/java/sdkman-multiple-versions-java-sbt-scala/)... if you're using a Mac at least.

Not sure about the Docker setup, but that shouldn't be needed.

Just just Java 8, install SBT, and you should be fine running `sbt test`.

**Comment by Nestor10** [comment]
Still needed or wanted? 
When i first looked at this I wasn't sure if schema was persevered when casting a df to a ds. But it is 
``` scala
    val ds1 = Seq(("x", 1)).toDF("a","b").as[tester]
    val ds2 = Seq((1, "x")).toDF("b","a").as[tester]
    ds1.printSchema()
    ds2.printSchema()
    println(
      ds1.collect() sameElements ds2.collect()
    ) //true
    assert(ds1.schema == ds2.schema ) //false
```

It does start to present a challenge, in that we have a two by two factorial now, col ordering -> yes/no row ordering -> yes/no. Also col ordering is probably incompatible with ignore column names.

**Comment by Nestor10** [comment]
https://github.com/Nestor10/spark-fast-tests/commit/1b767723e07052cb3419af3d046400996c8e32f3
A demo with working test. There are some weird things with encoders here, requiring us to be a little sneaky with datasets. But it work all new and old test work. Beyond how the demo code looks, do the ergonomics jive? Aka. Is this how we want this library to be used with one additional parameter.

**Comment by zeotuan** [comment]
Closing this Issue as it has already been implemented in #132 and will possibly be released in upcoming v1.0.1 #149
